spyder/app/tests/test_mainwindow.py

```diff
@@ -105,6 +126,42 @@ def close_window():
 # Tests
 #==============================================================================
 @flaky(max_runs=3)
+def test_kernel_connection(main_window, qtbot):
+    """Test that a kernel from Spyder is detected and attached to variable explorer."""
```

Comment:
Change this to

    Test that only Spyder kernels are connected to the Variable Explorer